code/modules/mining/machine_redemption.dm

```diff
@@ -23,6 +23,7 @@
 	var/datum/techweb/stored_research
 	var/obj/item/disk/design_disk/inserted_disk
 	var/datum/component/remote_materials/materials
+	var/direction_to_edit = 0 //1 is input, 2 is output, any other value is both (default) (should only ever be either 0, 1, or 2, but won't completely break if something weird happens and it becomes >2 or <0)
```

Comment:
Maybe switch this into a define?
e.g. have this in a defines file:

```dm
#define ORM_BOTH 0
#define ORM_INPUT 1
#define ORM_OUTPUT 2
```

and then have
```dm
var/direction_to_edit = ORM_BOTH
...
if(direction_to_edit == ORM_INPUT)
``` 
etc

other than that, code looks good

Reply:
Thanks for the pointer, I had a little voice in the back of my head telling me there was a better way to do this, just didn't know how. Should be implemented properly now, lemme know if I did something weird.